include/cocaine/context.hpp

```diff
@@ -44,37 +35,15 @@ namespace signal {
 class handler_base_t;
 }
 
-class actor_t;
-class execution_unit_t;
 
 class context_t {
     COCAINE_DECLARE_NONCOPYABLE(context_t)
 
-    typedef std::deque<std::pair<std::string, std::unique_ptr<actor_t>>> service_list_t;
-
-    // TODO: There was an idea to use the Repository to enable pluggable sinks and whatever else for
-    // for the Blackhole, when all the common stuff is extracted to a separate library.
-    std::unique_ptr<logging::logger_t> m_log;
-
-    // NOTE: This is the first object in the component tree, all the other dynamic components, be it
-    // storages or isolates, have to be declared after this one.
-    std::unique_ptr<api::repository_t> m_repository;
-
-    // A pool of execution units - threads responsible for doing all the service invocations.
-    std::vector<std::unique_ptr<execution_unit_t>> m_pool;
-
-    // Services are stored as a vector of pairs to preserve the initialization order. Synchronized,
-    // because services are allowed to start and stop other services during their lifetime.
-    synchronized<service_list_t> m_services;
-
-    // Context signalling hub.
-    retroactive_signal<io::context_tag> m_signals;
-
-public:
-    const config_t config;
-
-    // Service port mapping and pinning.
-    port_mapping_t mapper;
+    struct impl_t;
```

Comment:
I know, you have an allergy for "impl", but it's pimpl by design. context_data and other names will be misleading.